File: src/lessons/welcome/methods/slug/SlugHuntingEntity.c
//RemoteBuggle

void hunt();
int isFacingTrail();

void run(){
	hunt();
}


/* BEGIN TEMPLATE */
void hunt() {
/* BEGIN SOLUTION */
	while (! isOverBaggle()) {
		if (isFacingTrail()) {
			brushDown();
			forward(1);
			brushUp();
		} else {
			left();
		}
	}
	pickupBaggle();
}

int isFacingTrail(){
	if (isFacingWall())
		return 0;

	forward(1);
	int res = getGroundColor() == green;
	backward(1);
	return res;

/* END SOLUTION */
}

/* END TEMPLATE */

File: resources/exercises/methods/slug/SlugHuntingEntity.py
import java.awt.Color as Color

# BEGIN TEMPLATE
def hunt():
    # Complete this method
    # BEGIN SOLUTION
    while not isOverBaggle():
        brushUp()
        if isFacingTrail():
            brushDown()
            forward()
            brushUp()
        else:
            left()
    pickupBaggle()
def isFacingTrail():
   if isFacingWall():
      return False
   else:
	  forward()
	  res = getGroundColor() == Color.green
	  backward()
	  return res
# END SOLUTION

# and copy your isFacingTrail() over
# END TEMPLATE
hunt()
